### PR TITLE
add support for negative indexing of measurement numbers

### PR DESCRIPTION
**Pull request by pehses**
For example, the argument '-z -1' is now a shortcut for the last element of the multiMeasFile.

### CODE REVIEW

**Comment by hansenms** [comment]
Sorry for being late on this one. Trying to work my way through the backlog of issues and PR. This one we should either merge or close and it should have been done weeks ago.

I have some questions on this one. It feels slightly hacky to me. Like the only use would be "give me the last measurement because I don't care about calibration, and I don't actually know what is in my data". Is that the main use? To get the last measurement? Would you ever get the second to last? Also we have an option for all measurements, which @kspaceKelvin added. Would that suffice and just grab the last one? @kspaceKelvin do you see a need for this one?

One thing that makes this feel odd is that we don't use zero indexing here. That, for the record, was a mistake, but since we use 1 based indexing, -1 doesn't really feel like the last one. 

So bottom line is that I am a bit lukewarm on this one. But I am not totally against it either, so do you have super convincing argument here?

**Comment by hansenms** [comment]
We would at the very least need it to be updated to the latest master and I think we need to increment the minor version of converter since it sort of changes the API.

**Comment by pehses** [comment]
Yes, I think in most cases this feature would only be used to obtain the last measurement. The 7T data I'm working with often doesn't even contain more than one measurement in the raw data file. But I'm usually selecting the last measurement in the file anyway just in case. Using the 'all' option works for me too, though.

I should have probably explained my motivation for this commit a bit better: I watched the ISMRMRD tutorial at the ESMRMB's MRI Together workshop, and the presented way to select the last measurement was to select a huge measurement number (99), wait for siemens_to_ismrmrd to fail, and then to extract the number of measurements from the error message. This seemed very cumbersome to me. But this solution was probably invented before the 'all' option was added.

I agree, that it's a bit hacky in combination with 1-based indexing and I also agree that the 'all' option is in most cases a good alternative. The only downsides I see are that the conversion takes a bit more time than necessary and that the file size is a bit larger (the calibration scans are usually tiny, though, so this would only be relevant if you aren't interested in the last scan but in the second to last or so).

I don't insist on this PR, if you think that it adds unnecessary complexity then I'm fine with you closing it.

**Comment by hansenms** [comment]
@pehses makes sense. Thanks for the explanation. The "enter a large number and look at the error message" approach certainly is a bit awkward. Let's get this merged. Thanks for your contribution.

**Review by hansenms — APPROVED** [review]
